compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -301,7 +318,15 @@ impl Validator<'mir, 'tcx> {
 
         let mut err = op.build_error(self.ccx, span);
         assert!(err.is_error());
-        err.emit();
+
+        match op.importance() {
```

Comment:
As mentioned elsewhere, I think `build_error` should return a `Result<ErrorReported, DiagnosticBuilder<'tcx>>`. Additionally the `error_emitted` field could then be an `Option<ErrorReported>`

Reply:
Hmm... I just noticed that `emit` does not return `Option<ErrorReported>`.. sorry I thought we already had done all that. Let's carry on with this PR then and fix that together with making `emit` create instances of `ErrorReported`

Reply:
Sounds good. I like the `ErrorReported` approach.